### ISSUE TITLE
Question about data set

### ISSUE THREAD

**Issue by scott870430** [LiuYiwai/simple-IAM#2]
Hi! I am reading this paper, and your implementation allows me to better understand the idea of this paper, thanks!

And I have some questions about data set. Is your training data is from PASCAL VOC 2012 segmentation? Only use 1464 images for training, and 1449 images for validation?

Or you are using the "augmented pascal voc 2012 data set", which has 10582 images for training? If so, can you share how to get the augmented pascal voc 2012 data set with me?
I just only find [this](https://www.dropbox.com/s/oeu149j8qtbs1x0/SegmentationClassAug.zip?dl=0) , but i don't know where to get the raw image and other annotations.

By the way,the link to the Pascal voc 2012 data set is dead, it seems that the official website is broken

Thanks for your help! Have a nice day!

**Comment by LiuYiwai** [comment]
I'm sorry for my late reply. my PC has had problems these days, so I may not be able to provide a detailed answer, but I can tell you how I got the data set.

I downloaded the data set under the guidance of the PRM project. This is the previous work of this paper.

The difference between IAM and PRM is that IAM requires additional proposals as labels to train the filling module.

According to the description of the paper, proposals were obtained by MCG. I can get the data on the [official website](http://www.eecs.berkeley.edu/Research/Projects/CS/vision/grouping/mcg/) of MCG.

**Comment by LiuYiwai** [comment]
The obtained proposals seem to have weights, so I take the first 50 weights of each picture and convert them into a form that can be read by this project.

When I checked the number of proposal files, I found that there were 11540 files, so I used the same number of images in my training set, I hope this is helpful to you. If you want to get more information about the data set, maybe you can look at the PRM paper description.

Finally, on the issue of data download. If you cannot download the original data normally, the link may not work, or it may depend on your current region.

In my opinion, the best way to confirm is to use google colab service, save on google drive, and finally download back to local.

If you live in China and cannot use Google services well, I recommend a product called ShimakazeGo. It is an ACG-related auxiliary tool, but it can also help access Google and Y2B.

Hope to help you, if you have any questions, you can tell me.

**Comment by scott870430** [comment]
Hi! Thanks for your response!

Can I know which PRM project has the guidance for download the data set? I searched  [PRM](https://github.com/ZhouYanzhao/PRM/tree/pytorch) and [ultra-thin-PRM](https://github.com/chuchienshu/ultra-thin-PRM), but i didn't see it. You just use the data which describe in your readme?

About the data download, I use google colab service to download the data, it work!

Thanks for your help!

**Comment by LiuYiwai** [comment]
[PRM](https://github.com/ZhouYanzhao/PRM)
I found it in the second step of the Run demo section.

I'm glad to help you.

**Comment by LiuYiwai** [comment]
By the way, I only use the data in README, where the proposals data is part of my own processing. Maybe you'd like to reprocess it yourself and it may work better.

**Comment by scott870430** [comment]
Oh! I always skip that page.. Thanks!

I have roughly seen your code, and I have a question about model.
Are there any gradient update from Instance Extent Filling Module to PRM? The part I circled as shown below. 
https://i.imgur.com/Kdh8CPz.png

Expect the Feature Pyramid part, it seems that there is no gradient update in your implementation?
if no gradient update from IAM to PRM, can i modify model and reach that? 

I am confuse that is that is possible update gradient from IAM to PRM? (I have no idea how to do it) Or just I  don't know how to do it, but it is possible?
Or you actually do it but I miss it?

**Comment by LiuYiwai** [comment]
You are right, the Feature Pyramid part of my implementation does not backpropagate. I divided the PRM and Instance Extent Filling Module into two parts.

Because if you want to achieve the situation you describe, it seems that multiple labels will be backpropagated back to one input.

You know. For a picture, the PRM has one label, and the IAM part has multiple labels because there are multiple instances.

Maybe I couldn't solve this problem at that time.

Maybe this is the reason why my implementation is not so good :(

If you can have a better way to solve it, that's the best!

**Comment by scott870430** [comment]
> You are right, the Feature Pyramid part of my implementation does not backpropagate. I divided the PRM and Instance Extent Filling Module into two parts.
> 
> Because if you want to achieve the situation you describe, it seems that multiple tags will be backpropagated back to one input.
> 
> You know. For a picture, the PRM has one label, and the IAM part has multiple labels because there are multiple instances.
> 
> Maybe I couldn't solve this problem at that time.
> 
> Maybe this is the reason why my implementation is not so good :(
> 
> If you can have a better way to solve it, that's the best!

I think the Feature Pyramid part backpropagate implement may not too difficult..but the training strategy maybe has some problem?

Thanks for your help! Inspire me a lot!
